simple/lint.go

```diff
@@ -1839,3 +1840,55 @@ func (c *Checker) LintNilCheckAroundRange(j *lint.Job) {
 		ast.Inspect(f, fn)
 	}
 }
+
+func (c *Checker) LintSort(j *lint.Job) {
+	fn := func(node ast.Node) bool {
+		if !j.IsCallToAST(node, "sort.Sort") {
+			return true
+		}
+		call := node.(*ast.CallExpr)
+		if len(call.Args) == 0 {
+			return true
+		}
+
+		typeconv, ok := call.Args[0].(*ast.CallExpr)
+		if !ok {
+			return true
+		}
+		if len(typeconv.Args) == 0 {
+			return true
+		}
+
+		sel, ok := typeconv.Fun.(*ast.SelectorExpr)
```

Comment:
Instead of all of this code, I suggest using `lint.IsCallToAnyAST` and to change that function to return the function name that matched. Then this code simply becomes

```
name, ok := j.IsCallToAnyAST(typeconv, "sort.IntSlice", "sort.Float64Slice", "sort.StringSlice")
if !ok { return true }
```

Reply:
I couldn't change IsCallToAnyAST to check this case. It's meant for function calls, not type conversions. Even if I change that:

https://github.com/dominikh/go-tools/blob/4cd09d945fb3beeb0a478f1aa6522e1f03da362c/lint/lint.go#L608-L609

to a type switch, `types.TypeName` still doesn't have a method named `FullName`, and `Name` returns just `IntSlice` etc., not `sort.IntSlice`.

Reply:
OK